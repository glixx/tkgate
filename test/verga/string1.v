
module top;
  reg [7:0] a,b;
  reg [4:0] c,d;
  reg [10*8-1:0] x,y;
  integer i;

  initial
    begin
	a = 8'h42;

	{c,d} = a;
	$display("c=%h  d=%h",c,d);

	b[7:4] = 4'h6;
	b[3:0] = 4'h8;
	$display("b=%h",b);

	{a[3:0],a[7:4]} = a & 8'hff;
	$display("a=%h",a);

	x = "ABCDEFGHIJ";
    	$display("x=%s  x=%h",x,x);
 	for (i = 0;i < 10;i = i+1)
	  $display("x[%d:%d]=%s",8*i+7,8*i,x[8*i +: 8]);

 	for (i = 0;i < 10;i = i+1)
	  x[8*i +: 8] = (i+1)*(i+1);
	$display("x=%h",x);

 	for (i = 0;i < 10;i = i+1)
 	  begin	
	    y[8*i +: 8] = x[8*(9-i) +: 8];
	  end
	$display("y=%020h",y);

	// 
	// i should be read only on entering the loop so updates to i in
	// the body should not affect the number of iterations.
	// 
	i = 5;
	repeat (i)
	  begin
	    i = 2*(i - 1);
	    $display("hello i = %d",i);
	  end
	$display("done hello");
    end

endmodule
