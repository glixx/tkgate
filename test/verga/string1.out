c=2  d=2
b=68
a=24
x=ABCDEFGHIJ  x=4142434445464748494a
x[7:0]=J
x[15:8]=I
x[23:16]=H
x[31:24]=G
x[39:32]=F
x[47:40]=E
x[55:48]=D
x[63:56]=C
x[71:64]=B
x[79:72]=A
x=64514031241910090401
y=01040910192431405164
hello i = 8
hello i = 14
hello i = 26
hello i = 50
hello i = 98
done hello
